Userland/Libraries/LibWeb/Layout/FlexFormattingContext.cpp

```diff
@@ -1041,16 +1043,41 @@ void FlexFormattingContext::determine_hypothetical_cross_size_of_item(FlexItem&
     auto clamp_min = (!computed_min_size.is_auto() && (resolve_percentage_min_max_sizes || !computed_min_size.contains_percentage())) ? specified_cross_min_size(item.box) : 0;
     auto clamp_max = (!should_treat_cross_max_size_as_none(item.box) && (resolve_percentage_min_max_sizes || !computed_max_size.contains_percentage())) ? specified_cross_max_size(item.box) : CSSPixels::max();
 
-    // If we have a definite cross size, this is easy! No need to perform layout, we can just use it as-is.
+    // If we have a definite cross size, we may be able to use it as-is, if it is independent of the main size.
     if (has_definite_cross_size(item)) {
-        // To avoid subtracting padding and border twice for `box-sizing: border-box` only min and max clamp should happen on a second pass
-        if (resolve_percentage_min_max_sizes) {
-            item.hypothetical_cross_size = css_clamp(item.hypothetical_cross_size, clamp_min, clamp_max);
+        // We need to do layout again if the cross size depends on the main size. If not, we can skip it.
+        // Borrowed and adapted from Firefox:
+        // https://searchfox.org/mozilla-central/rev/f3e4b33a6122ce63bf81ae8c30cc5ac37458864b/layout/generic/nsFlexContainerFrame.cpp#2466-2512
```

Comment:
I'm not sure it's legally OK to include code in Ladybird that's derived from Firefox. We use different licenses, and I'm not a lawyer but BSD is more permissive than MPL is.

Reply:
I can just remove the optimization and add a TODO to re-add it, if that works

Reply:
That should be fine I think, yeah. Thanks!